### PR TITLE
Remove support for non-transformer based runs

### PR DESCRIPTION
**Pull request by mprobst**
This series of commits simplifies tsickle by removing support to *not* run in transformer mode. It also simplifies tsickle's API a bit and reduces the number of modes our tests run in.

### CODE REVIEW

File: test/golden_tsickle_test.ts [hunk 1]
```diff
@@ -75,28 +75,11 @@ function compareAgainstGolden(output: string|null, path: string) {
 const testFn = TEST_FILTER ? describe.only : describe;
 
 testFn('golden tests with TsickleCompilerHost', () => {
```
Comment (tbosch) [inline test/golden_tsickle_test.ts]: maybe `golden tests for tsickled sources` or something similar?

File: src/es5processor.ts [hunk 2]
```diff
@@ -25,11 +25,13 @@ export interface Es5ProcessorHost {
    * for each file.
    */
   fileNameToModuleId(fileName: string): string;
-}
-
-export interface Es5ProcessorOptions {
```
Comment (tbosch) [inline src/es5processor.ts]: +1 on merging the `options` with the `hosts`.

File: src/tsickle.ts [hunk 3]
```diff
@@ -1897,3 +1899,213 @@ export function getGeneratedExterns(externs: {[fileName: string]: string}): stri
   }
   return allExterns;
 }
+
+export interface TransformerHost extends es5processor.Es5ProcessorHost, AnnotatorHost {
```
Comment (tbosch) [inline src/tsickle.ts]: You could rename this back into `TsickleHost`..

File: test/golden_tsickle_test.ts (hunk 1 above)
Comment (tbosch) [reply test/golden_tsickle_test.ts]: Never mind, I saw that you changed it in a future commit.

File: src/tsickle.ts (hunk 3 above)
Comment (mprobst) [reply src/tsickle.ts]: Done.

File: src/tsickle.ts [hunk 4]
```diff
@@ -59,7 +57,7 @@ export enum AnnotatorFeatures {
    * TypeScript symbol information associated. Because of this, Typescript does not
    * elide exports for types in this case.
    *
-   * This flag will be removed and always set once we drop the on transformer version of tsickle.
+   * This flag will be removed and always set once we drop the on transformer version of
```
[the remark this answers is not in the document]
Comment (mprobst) [reply src/tsickle.ts]: Missed this, thanks for catching!

File: src/tsickle.ts [hunk 5]
```diff
@@ -1905,3 +1899,213 @@ export function getGeneratedExterns(externs: {[fileName: string]: string}): stri
...
+  externs: {[fileName: string]: string};
+}
+
+export interface EmitTransformers {
+  beforeTsickle?: Array<ts.TransformerFactory<ts.SourceFile>>;
+  beforeTs?: Array<ts.TransformerFactory<ts.SourceFile>>;
+  afterTs?: Array<ts.TransformerFactory<ts.SourceFile>>;
+}
+
+export function emitWithTsickle(
```
[the remark this answers is not in the document]
Comment (mprobst) [reply src/tsickle.ts]: see the commit description - it turns out `transformer.ts` and `tsickle.ts` had a cyclical value dependency. Given that we'll start removing features from `tsickle.ts` and moving them into their own transformer packages, I think this is the more sensible source organization: have the main transformation driver in `tsickle.ts`, and the individual transformers in their own files.

[the remark this answers is not in the document]
Comment (evmar) [reply src/tsickle.ts]: BTW to run tsickle in the TS playground we need to have no circular deps, so fixing this (by any mechanism, I didn't look at this one) is nice.

**Comment by mprobst** [comment]
Green in g3, committed.

**Pull request merged**